AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -60,30 +104,50 @@ extension QueryPredicate {
 }
 
 extension QueryPredicateOperation: GraphQLFilterConvertible {
-    var graphQLFilter: GraphQLFilter {
-        return [field: [self.operator.graphQLOperator: self.operator.value]]
+    func graphQLFilterInternal(_ modelSchema: ModelSchema? = nil) -> GraphQLFilter {
+        let filterValue = [self.operator.graphQLOperator: self.operator.value]
+        guard let modelSchema = modelSchema else {
+            return [field: filterValue]
+        }
+        return [columnName(modelSchema, forField: field): filterValue]
+    }
+
+    func columnName(_ modelSchema: ModelSchema, forField fieldName: String) -> String {
```

Comment:
if `field` is on the QueryPredicateOperation, then you don't need to pass it in to the extension method and just access it in the method itself:
```
guard let modelField = modelSchema.field(withName: field) else {
```

Reply:
in the latest code
```swift
func columnName(_ modelSchema: ModelSchema, forField fieldName: String) -> String {
    guard let modelField = modelSchema.field(withName: fieldName) else { // no need to pass in fieldName
            return field // you already have the field
```
you can also update the signature of the method to
`func columnName(_ modelSchema: ModelSchema)`